### PR TITLE
[libc] created integration tests for newhdrgen

### PR DESCRIPTION
**Pull request by aaryanshukla**
- created integration tests for libc hdrgen 
- implemented sorting function names in yaml files through script

### CODE REVIEW

File: libc/newhdrgen/tests/input/test_string.yaml [hunk 1]
```diff
@@ -0,0 +1,303 @@
+header: string.h
```
Comment (michaelrj-google) [inline libc/newhdrgen/tests/input/test_string.yaml:1]: for the test, you should make a much smaller yaml file, probably with a different name as well (so it's clear this is just for testing). You also need to make sure you're testing all of the parts, not just functioins. I'd recommend something like this:

```
header: test_header.h
macros: [MACRO_A, MACRO_B]
types: [type_a, type_b]
enums: [enum_a: [value_1, value_2], enum_b: [value_3]]
objects: [object_1, object_2]
functions: [list of functions]
```

File: libc/newhdrgen/tests/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import subprocess
+import unittest
+from pathlib import Path
+
+
+class TestHeaderGenIntegration(unittest.TestCase):
+    def setUp(self):
+        self.output_dir = Path("tests/output")
```
Comment (michaelrj-google) [inline libc/newhdrgen/tests/test_integration.py:8]: the output dir needs to be possible to set from an environment variable or command line argument if we want to use this with bazel.

File: libc/newhdrgen/yaml_to_classes.py [hunk 3]
```diff
@@ -46,11 +46,13 @@ def yaml_to_classes(yaml_data):
             Enumeration(enum_data["name"], enum_data.get("value", None))
         )
 
-    for function_data in yaml_data.get("functions", []):
+    functions = yaml_data.get("functions", [])
+    sorted_functions = sorted(functions, key=lambda x: x["name"])
+    for function_data in sorted_functions:
         arguments = [arg["type"] for arg in function_data["arguments"]]
         guard = function_data.get("guard", None)
         attributes = function_data.get("attributes", None)
```
Comment (SchrodingerZhu) [inline libc/newhdrgen/yaml_to_classes.py:54]: Notice that some attributes are positional dependent.

File: libc/newhdrgen/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+project(newhdrgen LANGUAGES CXX)
+
+enable_testing()
+
+set(NEWHDGEN_TESTS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/tests)
+
+add_test(
```
Comment (RoseZhang03) [inline libc/newhdrgen/CMakeLists.txt:7]: add only when in full build mode

Not sure if this is correct but: if(CMAKE_BUILD_TYPE STREQUAL FULL) /..../ endif()

Comment (aaryanshukla) [reply libc/newhdrgen/CMakeLists.txt:7]: Will fix!

File: libc/newhdrgen/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+if(LLVM_LIBC_FULL_BUILD)
```
Comment (michaelrj-google) [inline libc/newhdrgen/CMakeLists.txt:1]: if we've already checked for fullbuild when including this file we don't need to check it again here.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Pull request merged**